File: app/src/main/java/com/androidevlinux/percy/UTXO/data/network/apis/ChangellyApiImpl.java
package com.androidevlinux.percy.UTXO.data.network.apis;

import com.androidevlinux.percy.UTXO.data.models.changelly.GetCurrenciesResponseBean;
import com.androidevlinux.percy.UTXO.data.models.changelly.GetMinAmountReponseBean;
import com.androidevlinux.percy.UTXO.data.models.changelly.MainBodyBean;
import com.androidevlinux.percy.UTXO.data.models.changelly.TransactionBean;
import com.androidevlinux.percy.UTXO.utils.Constants;
import com.androidevlinux.percy.UTXO.utils.NativeUtils;

import retrofit2.Callback;

/**
 * Created by percy on 14/11/2017.
 */

public class ChangellyApiImpl extends AbstractBaseApi<ChangellyAPI> {
    private ChangellyAPI changellyAPI;
    private static ChangellyApiImpl changellyApiImpl;
    private ChangellyApiImpl(){
        setBaseUrl(NativeUtils.getChangellyBaseUrl());
        changellyAPI= getClient(ChangellyAPI.class);
    }

    public static ChangellyApiImpl getInstance(){
        if(changellyApiImpl ==null){
            changellyApiImpl = new ChangellyApiImpl();
        }
        return changellyApiImpl;
    }

    public void getCurrencies(String sign, MainBodyBean body, Callback<GetCurrenciesResponseBean> callback){
        changellyAPI.getCurrencies(CONTENT_TYPE, Constants.api_key,sign,body).enqueue(callback);
    }
    public void getMinAmount(String sign, MainBodyBean body, Callback<GetMinAmountReponseBean> callback){
        changellyAPI.getMinAmount(CONTENT_TYPE,Constants.api_key,sign,body).enqueue(callback);
    }

    public void createTransaction(String sign, MainBodyBean body, Callback<TransactionBean> callback){
        changellyAPI.createTransaction(CONTENT_TYPE,Constants.api_key,sign,body).enqueue(callback);
    }
}

File: app/src/main/java/com/androidevlinux/percy/UTXO/data/network/apis/ChangellyApiImpl.kt
package com.androidevlinux.percy.UTXO.data.network.apis

import com.androidevlinux.percy.UTXO.data.models.changelly.GetCurrenciesResponseBean
import com.androidevlinux.percy.UTXO.data.models.changelly.GetMinAmountReponseBean
import com.androidevlinux.percy.UTXO.data.models.changelly.MainBodyBean
import com.androidevlinux.percy.UTXO.data.models.changelly.TransactionBean
import com.androidevlinux.percy.UTXO.utils.NativeUtils
import retrofit2.Callback

/**
 * Created by percy on 14/11/2017.
 */

class ChangellyApiImpl private constructor() : AbstractBaseApi<ChangellyAPI>() {
    private val changellyAPI: ChangellyAPI

    init {
        setBaseUrl(NativeUtils.changellyBaseUrl)
        changellyAPI = getClient(ChangellyAPI::class.java)
    }

    fun getCurrencies(sign: String, body: MainBodyBean, callback: Callback<GetCurrenciesResponseBean>) {
        changellyAPI.getCurrencies(contentType, NativeUtils.changellyApiKey, sign, body).enqueue(callback)
    }

    fun getMinAmount(sign: String, body: MainBodyBean, callback: Callback<GetMinAmountReponseBean>) {
        changellyAPI.getMinAmount(contentType, NativeUtils.changellyApiKey, sign, body).enqueue(callback)
    }

    fun createTransaction(sign: String, body: MainBodyBean, callback: Callback<TransactionBean>) {
        changellyAPI.createTransaction(contentType, NativeUtils.changellyApiKey, sign, body).enqueue(callback)
    }

    companion object {
        private var changellyApiImpl: ChangellyApiImpl? = null

        val instance: ChangellyApiImpl
            get() {
                if (changellyApiImpl == null) {
                    changellyApiImpl = ChangellyApiImpl()
                }
                return changellyApiImpl!!
            }
    }
}

